test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/CustomUriFunctionsTests.cs

```diff
@@ -509,6 +510,141 @@ FunctionSignatureWithReturnType myStringFunction
             }
         }
 
+        [Fact]
+        public void ParseWithMixedCaseCustomUriFunction_EnableCaseInsensitive_ShouldWork()
+        {
+            try
+            {
+                FunctionSignatureWithReturnType myStringFunction
+                    = new FunctionSignatureWithReturnType(EdmCoreModel.Instance.GetBoolean(true), EdmCoreModel.Instance.GetString(true), EdmCoreModel.Instance.GetString(true));
+
+                // Add a custom uri function
+                CustomUriFunctions.AddCustomUriFunction("myFirstMixedCasestringfunction", myStringFunction);
+
+                // Uri with mixed-case, should work for resolver with case insensitive enabled.
+                var fullUri = new Uri("http://www.odata.com/OData/People" + "?$filter=mYFirstMixedCasesTrInGfUnCtIoN(Name, 'BlaBla')");
+                ODataUriParser parser = new ODataUriParser(HardCodedTestModel.TestModel, new Uri("http://www.odata.com/OData/"), fullUri);
+                parser.Resolver.EnableCaseInsensitive = true;
+
+                var startsWithArgs = parser.ParseFilter().Expression.ShouldBeSingleValueFunctionCallQueryNode("myFirstMixedCasestringfunction")
+                    .And.Parameters.ToList();
+                startsWithArgs[0].ShouldBeSingleValuePropertyAccessQueryNode(HardCodedTestModel.GetPersonNameProp());
+                startsWithArgs[1].ShouldBeConstantQueryNode("BlaBla");
+            }
+            finally
+            {
+                CustomUriFunctions.RemoveCustomUriFunction("myFirstMixedCasestringfunction").Should().BeTrue();
+            }
+        }
+
+        [Fact]
+        public void ParseWithExactMatchCustomUriFunction_EnableCaseInsensitive_ShouldWorkForMultipleEquiventArgumentsMatches()

```

Comment:
typo: 
ShouldWorkForMultipleEquiventArgumentsMatches
ShouldWorkForMultipleEquiv**al**entArgumentsMatches

Reply:
good catch. Thanks.

---
In reply to: [190790144](https://github.com/OData/odata.net/pull/1156#discussion_r190790144) [](ancestors = 190790144)